File: cpp/questions/maxProfit.cpp
#include <bits/stdc++.h>

using namespace std;

class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int mv = prices[0];
        int mp = 0;
        for (const auto& p: prices) {
            mv = min(mv, p);
            mp = max(mp, p-mv);
        }
        return mp;
    }
};

File: rust/questions/max_profit.rs
impl Solution {
    pub fn max_profit(prices: Vec<i32>) -> i32 {
        let mut mv = prices[0];
        let mut mp = 0;
        for p in prices {
            mv = mv.min(p);
            mp = mp.max(p - mv);
        }
        return mp;
    }
}
